src/modules/rlm_mschap/auth_wbclient.c

```diff
@@ -147,6 +181,51 @@ int do_auth_wbclient(rlm_mschap_t *inst, REQUEST *request,
 		REDEBUG2("Domain not found");
 		break;
 	case WBC_ERR_AUTH_ERROR:
+		if (inst->wb_retry_with_normalized_username) {
```

Comment:
Rather than this chunk of code here, if it is put in before the switch with just a simple
```
if (err == WBC_ERR_AUTH_ERROR && inst->wb_retry_with_normalized_username) {
  ...
}
```
then the existing err handling and useful error messages will be displayed after a username rewrite.

Otherwise if the actual error results in e.g. password expired, that will be missed if the auth is retried.

You can then also leave the fr_connection_release before the switch (and after this if() ), avoiding the other issue I just wrote.